ios/Plugin/KlippaScannerSDKPlugin.swift

```diff
@@ -15,235 +15,313 @@ public class KlippaScannerSDKPlugin: CAPPlugin {
+                single.instructions = Instructions(
+                    message: instructions,
+                    dismissHandler: { [weak self] in
+                        self?.singleDocumentModeInstructionsDismissed = true
+                    }
+                )
+            }
+            modes.append(single)
+        }
+
+        if let multiCameraMode = config["cameraModeMulti"] as? [String: String] {
+            let multi = KlippaMultipleDocumentMode()
+
+            if let name = multiCameraMode["name"] {
+                multi.name = name
+            }
+
+            if let instructions = multiCameraMode["message"] {
+                multi.instructions = Instructions(
+                    message: instructions,
+                    dismissHandler: { [weak self] in
+                        self?.multiDocumentModeInstructionsDismissed = true
+                    }
+                )
+            }
+            modes.append(multi)
+        }
+
+        if let segmentedCameraMode = config["cameraModeSegmented"] as? [String: String] {
+            let segmented = KlippaSegmentedDocumentMode()
+
+            if let name = segmentedCameraMode["name"] {
+                segmented.name = name
+            }
+
+            if let instructions = segmentedCameraMode["message"] {
+                segmented.instructions = Instructions(
+                    message: instructions,
+                    dismissHandler: { [weak self] in
+                        self?.multiDocumentModeInstructionsDismissed = true
+                    }
+                )
+            }
+            modes.append(segmented)
+        }
+
+        if modes.isEmpty {
+            return
+        }
+
+        let index = config["startingIndex"] as? Int ?? 0
+
+        builder.klippaCameraModes = KlippaCameraModes(
+            modes: modes,
+            startingIndex: index
+        )
+    }
+
     //MARK: Setup Messages
-    fileprivate func setupMessages(_ config: [AnyHashable : Any]) {
+    private func setupMessages(_ config: [AnyHashable : Any], _ builder: KlippaScannerBuilder) {
         if let imageTooBrightMessage = config["imageTooBrightMessage"] as? String {
-            KlippaScanner.setup.imageTooBrightMessage = imageTooBrightMessage
+            builder.klippaMessages.imageTooBrightMessage = imageTooBrightMessage
         }
 
         if let imageTooDarkMessage = config["imageTooDarkMessage"] as? String {
-            KlippaScanner.setup.imageTooDarkMessage = imageTooDarkMessage
+            builder.klippaMessages.imageTooDarkMessage = imageTooDarkMessage
         }
 
         if let imageColorOriginalText = config["imageColorOriginalText"] as? String {
-            KlippaScanner.setup.imageColorOriginalText = imageColorOriginalText
+            builder.klippaButtonTexts.imageColorOriginalText = imageColorOriginalText
         }
 
         if let imageColorGrayscaleText = config["imageColorGrayscaleText"] as? String {
```

Comment:
```suggestion
        if let imageColorGrayscaleText = config["imageColorGrayScaleText"] as? String {
```

I think, `imageColorGrayscaleText` should be capitalised here and `index.tsx` file as well.

Reply:
Fixed.